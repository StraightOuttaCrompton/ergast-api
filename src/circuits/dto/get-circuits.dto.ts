import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";

export class GetCircuitsDto {
    /**
     * TODO: extract into class which this inherits from
     */
    @IsOptional()
    @IsInt()
    @IsPositive()
    offset: number;

    @IsOptional()
    @IsInt()
    @IsPositive()
    limit: number;
    /**
     * --------------------------------------------------
     */

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
