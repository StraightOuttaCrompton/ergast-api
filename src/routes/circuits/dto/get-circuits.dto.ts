import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetCircuitsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
