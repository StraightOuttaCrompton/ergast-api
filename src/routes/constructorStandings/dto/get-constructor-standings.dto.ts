import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
