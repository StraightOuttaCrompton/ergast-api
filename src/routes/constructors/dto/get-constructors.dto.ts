import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number; // TODO: validate for actual status values

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    constructorStandings?: number;
}
