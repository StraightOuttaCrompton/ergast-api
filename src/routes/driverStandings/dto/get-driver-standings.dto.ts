import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriverStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
