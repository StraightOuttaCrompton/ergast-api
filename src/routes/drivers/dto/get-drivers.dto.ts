import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriversDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    driverStandings?: number;
}
