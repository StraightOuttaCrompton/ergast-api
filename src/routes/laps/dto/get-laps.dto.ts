import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetLapsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;
}
