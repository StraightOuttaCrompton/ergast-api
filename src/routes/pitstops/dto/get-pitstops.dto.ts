import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetPitstopsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    lapNumber: number;
}
