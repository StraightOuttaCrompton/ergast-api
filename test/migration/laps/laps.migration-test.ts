import querystring from "querystring";
import { getMigrationTest } from "../migrationUtils";
import { GetLapsDto } from "src/routes/laps/dto/get-laps.dto";
import path from "path";

describe("GET /laps/${lapNumber} smoke tests", () => {
    const migrationTest = getMigrationTest(__dirname, path.basename(__filename), (response) => {
        const { season, round, ...rest } = response.MRData.RaceTable.Races[0];

        return rest;
    });

    migrationTest(
        `/laps/1?${querystring.stringify({ year: 2022, round: 8 })}`,
        `https://ergast.com/api/f1/2022/8/laps/1.json`
    );
});

describe("GET /laps smoke tests", () => {
    const migrationTest = getMigrationTest(__dirname, path.basename(__filename), (response) => {
        const { season, round, ...rest } = response.MRData.RaceTable.Races[0];

        return rest;
    });
    const endpoint = "/laps";

    const queries: { queryParams: Partial<GetLapsDto>; legacyRoute: string }[] = [
        {
            queryParams: { year: 2022, round: 3 },
            legacyRoute: "2022/3",
        },
        {
            queryParams: { year: 2011, round: 5, driverId: "alonso" },
            legacyRoute: "2011/5/drivers/alonso",
        },
    ];

    queries.forEach(({ queryParams, legacyRoute }) => {
        migrationTest(
            `${endpoint}?${querystring.stringify(queryParams)}`,
            `https://ergast.com/api/f1/${legacyRoute}/laps.json`
        );
    });
});
