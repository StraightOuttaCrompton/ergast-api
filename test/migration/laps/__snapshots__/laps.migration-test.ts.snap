// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /laps smoke tests /laps?year=2011&round=5&driverId=alonso 1`] = `
{
  "Circuit": {
    "Location": {
      "country": "Spain",
      "lat": "41.57",
      "locality": "Montmel√≥",
      "long": "2.26111",
    },
    "circuitId": "catalunya",
    "circuitName": "Circuit de Barcelona-Catalunya",
    "url": "http://en.wikipedia.org/wiki/Circuit_de_Barcelona-Catalunya",
  },
  "Laps": [
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:34.494",
        },
      ],
      "number": "1",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.812",
        },
      ],
      "number": "2",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.606",
        },
      ],
      "number": "3",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.012",
        },
      ],
      "number": "4",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.318",
        },
      ],
      "number": "5",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.039",
        },
      ],
      "number": "6",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.007",
        },
      ],
      "number": "7",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.164",
        },
      ],
      "number": "8",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.303",
        },
      ],
      "number": "9",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:32.825",
        },
      ],
      "number": "10",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "2",
          "time": "1:48.222",
        },
      ],
      "number": "11",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:29.323",
        },
      ],
      "number": "12",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:29.809",
        },
      ],
      "number": "13",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:29.778",
        },
      ],
      "number": "14",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:29.856",
        },
      ],
      "number": "15",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:29.898",
        },
      ],
      "number": "16",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.316",
        },
      ],
      "number": "17",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "1",
          "time": "1:30.475",
        },
      ],
      "number": "18",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "2",
          "time": "1:32.615",
        },
      ],
      "number": "19",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:45.987",
        },
      ],
      "number": "20",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:28.878",
        },
      ],
      "number": "21",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:28.737",
        },
      ],
      "number": "22",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:29.003",
        },
      ],
      "number": "23",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:29.670",
        },
      ],
      "number": "24",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:29.332",
        },
      ],
      "number": "25",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:29.538",
        },
      ],
      "number": "26",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:29.543",
        },
      ],
      "number": "27",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:29.828",
        },
      ],
      "number": "28",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "3",
          "time": "1:32.309",
        },
      ],
      "number": "29",
    },
    {
      "Timings": [
        {
          "driverId": "alonso",
          "position": "4",
          "time": "1:48.274",
        },
      ],
      "number": "30",
    },
  ],
  "date": "2011-05-22",
  "raceName": "Spanish Grand Prix",
  "time": "12:00:00Z",
  "url": "http://en.wikipedia.org/wiki/2011_Spanish_Grand_Prix",
}
`;

exports[`GET /laps smoke tests /laps?year=2022&round=3 1`] = `
{
  "Circuit": {
    "Location": {
      "country": "Australia",
      "lat": "-37.8497",
      "locality": "Melbourne",
      "long": "144.968",
    },
    "circuitId": "albert_park",
    "circuitName": "Albert Park Grand Prix Circuit",
    "url": "http://en.wikipedia.org/wiki/Melbourne_Grand_Prix_Circuit",
  },
  "Laps": [
    {
      "Timings": [
        {
          "driverId": "leclerc",
          "position": "1",
          "time": "1:29.695",
        },
        {
          "driverId": "max_verstappen",
          "position": "2",
          "time": "1:30.342",
        },
        {
          "driverId": "hamilton",
          "position": "3",
          "time": "1:31.296",
        },
        {
          "driverId": "perez",
          "position": "4",
          "time": "1:32.265",
        },
        {
          "driverId": "russell",
          "position": "5",
          "time": "1:33.325",
        },
        {
          "driverId": "norris",
          "position": "6",
          "time": "1:34.050",
        },
        {
          "driverId": "ricciardo",
          "position": "7",
          "time": "1:34.558",
        },
        {
          "driverId": "ocon",
          "position": "8",
          "time": "1:35.268",
        },
        {
          "driverId": "gasly",
          "position": "9",
          "time": "1:36.461",
        },
        {
          "driverId": "alonso",
          "position": "10",
          "time": "1:37.135",
        },
        {
          "driverId": "tsunoda",
          "position": "11",
          "time": "1:37.589",
        },
        {
          "driverId": "bottas",
          "position": "12",
          "time": "1:38.109",
        },
        {
          "driverId": "mick_schumacher",
          "position": "13",
          "time": "1:38.502",
        },
        {
          "driverId": "sainz",
          "position": "14",
          "time": "1:39.028",
        },
        {
          "driverId": "zhou",
          "position": "15",
          "time": "1:39.232",
        },
        {
          "driverId": "kevin_magnussen",
          "position": "16",
          "time": "1:39.709",
        },
        {
          "driverId": "vettel",
          "position": "17",
          "time": "1:40.136",
        },
        {
          "driverId": "latifi",
          "position": "18",
          "time": "1:40.840",
        },
        {
          "driverId": "stroll",
          "position": "19",
          "time": "1:41.235",
        },
        {
          "driverId": "albon",
          "position": "20",
          "time": "1:41.690",
        },
      ],
      "number": "1",
    },
    {
      "Timings": [
        {
          "driverId": "leclerc",
          "position": "1",
          "time": "1:24.746",
        },
        {
          "driverId": "max_verstappen",
          "position": "2",
          "time": "1:24.831",
        },
        {
          "driverId": "hamilton",
          "position": "3",
          "time": "1:25.455",
        },
        {
          "driverId": "perez",
          "position": "4",
          "time": "1:25.290",
        },
        {
          "driverId": "russell",
          "position": "5",
          "time": "1:25.459",
        },
        {
          "driverId": "norris",
          "position": "6",
          "time": "1:25.942",
        },
        {
          "driverId": "ricciardo",
          "position": "7",
          "time": "1:26.362",
        },
        {
          "driverId": "ocon",
          "position": "8",
          "time": "1:26.734",
        },
        {
          "driverId": "gasly",
          "position": "9",
          "time": "1:26.622",
        },
        {
          "driverId": "alonso",
          "position": "10",
          "time": "1:27.059",
        },
      ],
      "number": "2",
    },
  ],
  "date": "2022-04-10",
  "raceName": "Australian Grand Prix",
  "time": "05:00:00Z",
  "url": "http://en.wikipedia.org/wiki/2022_Australian_Grand_Prix",
}
`;

exports[`GET /laps/\${lapNumber} smoke tests /laps/1?year=2022&round=8 1`] = `
{
  "Circuit": {
    "Location": {
      "country": "Azerbaijan",
      "lat": "40.3725",
      "locality": "Baku",
      "long": "49.8533",
    },
    "circuitId": "baku",
    "circuitName": "Baku City Circuit",
    "url": "http://en.wikipedia.org/wiki/Baku_City_Circuit",
  },
  "Laps": [
    {
      "Timings": [
        {
          "driverId": "perez",
          "position": "1",
          "time": "1:50.927",
        },
        {
          "driverId": "leclerc",
          "position": "2",
          "time": "1:52.264",
        },
        {
          "driverId": "max_verstappen",
          "position": "3",
          "time": "1:52.920",
        },
        {
          "driverId": "sainz",
          "position": "4",
          "time": "1:54.359",
        },
        {
          "driverId": "russell",
          "position": "5",
          "time": "1:55.452",
        },
        {
          "driverId": "gasly",
          "position": "6",
          "time": "1:56.234",
        },
        {
          "driverId": "hamilton",
          "position": "7",
          "time": "1:56.952",
        },
        {
          "driverId": "vettel",
          "position": "8",
          "time": "1:57.643",
        },
        {
          "driverId": "tsunoda",
          "position": "9",
          "time": "1:58.400",
        },
        {
          "driverId": "alonso",
          "position": "10",
          "time": "1:58.543",
        },
        {
          "driverId": "norris",
          "position": "11",
          "time": "1:59.787",
        },
        {
          "driverId": "ricciardo",
          "position": "12",
          "time": "2:00.530",
        },
        {
          "driverId": "ocon",
          "position": "13",
          "time": "2:00.803",
        },
        {
          "driverId": "zhou",
          "position": "14",
          "time": "2:01.528",
        },
        {
          "driverId": "albon",
          "position": "15",
          "time": "2:02.720",
        },
        {
          "driverId": "bottas",
          "position": "16",
          "time": "2:02.721",
        },
        {
          "driverId": "stroll",
          "position": "17",
          "time": "2:03.829",
        },
        {
          "driverId": "kevin_magnussen",
          "position": "18",
          "time": "2:03.859",
        },
        {
          "driverId": "latifi",
          "position": "19",
          "time": "2:04.909",
        },
        {
          "driverId": "mick_schumacher",
          "position": "20",
          "time": "2:05.761",
        },
      ],
      "number": "1",
    },
  ],
  "date": "2022-06-12",
  "raceName": "Azerbaijan Grand Prix",
  "time": "11:00:00Z",
  "url": "http://en.wikipedia.org/wiki/2022_Azerbaijan_Grand_Prix",
}
`;
